pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,297 @@
+						}
+					}
+				}
+			}
+		}
+	}
+	return res, nil
+}
+
+// getLocalProfileConfig connects to local profile server to fetch current profile
+func getLocalProfileConfig(localServerURL string, iteration int, getconfigCtx *getconfigContext) (*profile.LocalProfile, error) {
+
+	log.Tracef("getLocalProfileConfig(%s, %d)", localServerURL, iteration)
+
+	localServerMap, err := prepareLocalProfileServerMap(localServerURL, getconfigCtx)
+	if err != nil {
+		return nil, fmt.Errorf("getLocalProfileConfig: prepareLocalProfileServerMap: %s", err)
+	}
+
+	if len(localServerMap) == 0 {
+		return nil, fmt.Errorf(
+			"getLocalProfileConfig: cannot find any configured local subnets for localServerURL: %s",
+			localServerURL)
+	}
+
+	var errList []string
+	for bridgeName, urlAndSrcIPs := range localServerMap {
+		for _, el := range urlAndSrcIPs {
+			resp, contents, err := zedcloud.SendLocal(zedcloudCtx, el.actualURL, bridgeName, el.srcIP, 0, nil)
+			if err != nil {
+				errList = append(errList, fmt.Sprintf("SendLocal: %s", err))
+				continue
+			}
+			if resp.StatusCode != http.StatusOK {
+				errList = append(errList, fmt.Sprintf("SendLocal: wrong response status code: %d",
+					resp.StatusCode))
+				continue
+			}
+			if err := validateProtoMessage(el.actualURL, resp); err != nil {
+				// send something to ledmanager ???
+				errList = append(errList, fmt.Sprintf("validateProtoMessage: resp header error: %s", err))
+				continue
+			}
+			localProfile, err := parseAndValidateLocalProfile(contents, getconfigCtx)
+			if err != nil {
+				errList = append(errList, fmt.Sprintf("parseAndValidateLocalProfile: %s", err))
+				continue
+			}
+			return localProfile, nil
+		}
+	}
+	return nil, fmt.Errorf("getLocalProfileConfig: all attempts failed: %s", strings.Join(errList, ";"))
+}
+
+func writeReceivedProtoMessageLocalProfile(localProfile *profile.LocalProfile) {
+	contents, err := proto.Marshal(localProfile)
+	if err != nil {
+		log.Errorf("writeReceivedProtoMessageLocalProfile Marshalling failed: %s", err)
+		return
+	}
+	writeProtoMessage(savedLocalProfileFile, contents)
+	return
+}
+
+//parseProfile process local and global profile configuration
+//must be called before processing of app instances from config
+//if usingSaved will try to read saved localProfile
+func parseProfile(ctx *getconfigContext, config *zconfig.EdgeDevConfig, usingSaved bool) {
+	log.Functionf("parseProfile start: globalProfile: %s currentProfile: %s",
+		ctx.globalProfile, ctx.currentProfile)
+	ctx.globalProfile = config.GlobalProfile
+	if ctx.localProfileServer != config.LocalProfileServer {
+		if !ctx.localProfileServerSet {
+			// first time set, do nothing
+			ctx.localProfileServerSet = true
+		} else {
+			log.Noticef("parseProfile: LocalProfileServer changed from %s to %s",
+				ctx.localProfileServer, config.LocalProfileServer)
+			// if we change localProfileServer, remove saved profile
+			cleanSavedProtoMessage(savedLocalProfileFile)
+		}
```

Comment:
I don't understand the need for this complexity. We don't need to cleanSaved  when we have received an update, because in that state we will write/save the update.

But we need to read the saved as we start the timer task.